pkg/storage/db.go

```diff
@@ -215,3 +218,58 @@ func validateReceiverSchema(schema string, eventPayload types.JSON) error {
 
 	return nil
 }
+
+//go:embed megaQuery.sql
+var megaQuery string
+
+// Data represents the EventReceiverGroup data that comes back from
+// the mega query. It is necessary to allow us to automatically
+// insert event_receiver_ids by overriding the json tag
+type Data struct {
+	EventReceiverGroup
+
+	// Database returns json array instead of a string
+	EventReceiverIDs types.JSON `json:"event_receiver_ids"`
+}
+
+func FindTriggeredEventReceiverGroups(tx *gorm.DB, event Event, eventReceiverID graphql.ID) ([]EventReceiverGroup, error) {
+	var data []Data
+	result := tx.Model("EventReceiverGroup").Raw(megaQuery,
+		sql.Named("name", event.Name),
+		sql.Named("version", event.Version),
+		sql.Named("release", event.Release),
+		sql.Named("platform_id", event.PlatformID),
+		sql.Named("package", event.Package),
+		sql.Named("event_receiver_id", eventReceiverID)).
+		Scan(&data)
+	if result.Error != nil {
+		if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+			return nil, fmt.Errorf("eventReceiverGroup %s not found", eventReceiverID)
+		}
+		return nil, pgError(result.Error)
+	}
+	if result.RowsAffected < 1 {
```

Comment:
I'd prefer checking == 0 instead of < 1, unless RowsAffected is ever expected to have a negative value.

Reply:
I don't expect it necessarily, but I'd rather have us covered just in case. I don't think its more expensive for us to check the length like this here